### ISSUE TITLE
Support for media keys/shortcuts

### ISSUE THREAD

**Issue by umpirsky** [nukeop/nuclear#18]
![screenshot from 2017-02-24 12-07-31](https://cloud.githubusercontent.com/assets/208957/23301360/d5da980e-fa89-11e6-84e1-c35dcd4105d3.png)

**Comment by umpirsky** [comment]
Installed latest version 0.3.0. Media keys still don't work.

**Comment by nukeop** [comment]
Since these are global shortcuts, if you have them assigned to anything else, Electron won't be able to use them. Maybe that's the case?

If not, I'll take another look at this.

**Comment by umpirsky** [comment]
@nukeop No, they work with other media players (e.g. Rhytmbox). Thanks.

**Comment by nukeop** [comment]
By the way, what's your system, and how did you install the program?

**Comment by umpirsky** [comment]
Version:

```
 lsb_release -a

Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial
```

```
gnome-shell --version
GNOME Shell 3.18.5
```

I installed latest deb package https://github.com/nukeop/nuclear/releases/download/george/nuclear-george-linux-x64.deb.

**Comment by nukeop** [comment]
Can you try using the dev version?

Then in `app/main.development.js`, please add the following lines after updater.init:

```
  var registered = globalShortcut.register('mediaplaypause', function () {
    console.log('mediaplaypause pressed');
  });
  if (!registered) {
    console.log('mediaplaypause registration failed');
  } else {
    console.log('mediaplaypause registration bound!');
}
```

And add `globalShortcut` to electron imports (first line).
If you run it then, we'll be able to see if there's something preventing you from binding media keys. Any errors will appear in the terminal.

**Comment by umpirsky** [comment]
@nukeop How can i run dev version?

**Comment by fcastilloec** [comment]
Hi, I ran the code you posted and I get "mediaplaypause registration failed"
If I switch the to any other keyword or just add ctrl to the mediaplaypause, it works correctly.

I found a similar issue in electron itself [electron issue #3600](https://github.com/electron/electron/issues/3600)

So it seems that electron can't use already assigned global keywords for its application.

**Comment by nukeop** [comment]
Yeah, this is true.
This will be solved more robustly in 0.4.0.

**Comment by fcastilloec** [comment]
@nukeop As a suggestion, you could look at (Google-Play-Music-Desktop-Player-UNOFFICIAL-/mediaKeysDBus.js)[https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/master/src/main/features/linux/mediaKeysDBus.js] to implement a `dbus` interface and fix the issue on Linux.
The media keys are always bounded by the OS on Linux, so you just need to communicate through `dbus`

**Comment by nukeop** [comment]
Nice, I'll have to look into dbus to implement MPRIS in the new version as well.

**Comment by GNOME-IS-LIFE** [comment]
Any update?

**Comment by nukeop** [comment]
Yeah, mpris support is implemented in 0.4.0 (rewrite branch and nuclear-rewrite repo).

**Comment by nukeop** [comment]
Since dbus works and mpris is looking good too I can start looking at handling media keys.

**Comment by sergio-rojasa** [comment]
Is anyone working on this issue?
If not then I would like to work on it?

**Comment by forteller** [comment]
Hi. I just tried Nuclear again for the first time in a long time, and I'm so happy with it! Really great work! But unfortunately the media keys on my keyboard does not work with it, although the work with other media players. I installed the flatpak, but my other media players is also installed through flatpak, so I don't think that should be the issue? 

I'm on Fedora Silverblue 44. Thank you so much!
